homeassistant/components/lg_netcast/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow to configure the LG Netcast TV integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pylgnetcast import AccessTokenError, LgNetCastClient, SessionIdError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST, CONF_ID, CONF_NAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.util.network import is_host_valid
+
+from .const import DEFAULT_NAME, DOMAIN
+from .helpers import LGNetCastDetailDiscoveryError, async_discover_netcast_details
+
+
+class LGNetCast(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for LG Netcast TV integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.client: LgNetCastClient | None = None
+        self.device_config: dict[str, Any] = {}
+        self._discovered_devices: dict[str, Any] = {}
+
+    def create_client(self) -> None:
+        """Create LG Netcast client from config."""
+        host = self.device_config[CONF_HOST]
+        access_token = self.device_config.get(CONF_ACCESS_TOKEN)
+        self.client = LgNetCastClient(host, access_token)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            if is_host_valid(host):
+                self.device_config[CONF_HOST] = host
+                return await self.async_step_authorize()
+
+            errors[CONF_HOST] = "invalid_host"
+
+        user_input = user_input or {}
```

Comment:
Why is this statement needed?

Reply:
Good spot, it's left over from the auto discovery logic that I removed from the initial commit, where `CONF_HOST` becomes an optional field and I was setting the default based on the value from `user_input` if available (otherwise it's empty). But it is unnecessary for the basic config flow implementation. I will remove it

Reply:
This line is now removed